### PR TITLE
#2 - Add training of existing models/scratch models

### PR DESCRIPTION
**Pull request by bjtulinl**
- Add training of existing ner and pos models

### CODE REVIEW

**Comment by jcklie** [comment]
Why do you not create a Corpus on the fly instead of writing to a file and loading it? There is `TaggedCorpus` which just needs tagged flair sentences.

**Comment by jcklie** [comment]
What I mean is:
1. You create flair tokens for every sentence: https://github.com/zalandoresearch/flair/blob/master/flair/data.py#L170
2. You add these tokens to sentences: https://github.com/zalandoresearch/flair/blob/master/flair/data.py#L435
3. You add create a sentence dataset https://github.com/zalandoresearch/flair/blob/477d4004b4029efdd5af140bbb928101c305033e/flair/datasets.py#L356
4. You add these to a corpus: https://github.com/zalandoresearch/flair/blob/master/flair/data.py#L747

That is what flair internally is doing when you load a file

File: app_flair.py [hunk 1]
```diff
@@ -96,6 +123,103 @@ def parse_prediction_request(json_object: JsonDict) -> PredictionRequest:
     return PredictionRequest(layer, feature, projectId, Document(xmi, documentId, userId), typesystem)
 
 
+def dataset_split(list_sentences: list):
+    l = len(list_sentences)
+    test_size: int = round(l / 10)
+    dev_size: int = round(l / 10)
+    random.shuffle(list_sentences)
```
Comment (jcklie) [inline app_flair.py]: I think you are shuffling the wrong list here, you should not mutate input parameters. You can also just use sklearn: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html

**Comment by jcklie** [comment]
@bjtulinl Can you fix this merge conflict?

**Comment by bjtulinl** [comment]
Ok. I will do that.

File: app_flair.py (hunk 1 above)
Comment (jcklie) [reply app_flair.py]: @bjtulinl Did you have a look at this?

Comment (bjtulinl) [reply app_flair.py]: Yes. But it seems that the function train_test_split in sklearn is used to split the dataset with the labels. In flair they use random_split in torch. So I write the function like this:
```
def dataset_split(list_sentences: list):
    l = len(list_sentences)
    dev_size: int = round(l / 10)
    train, dev = random_split(list_sentences,[l-dev_size,dev_size])
    train_l = len(train)
    test_size: int = round(train_l / 10)
    train, test = random_split(train, [train_l-test_size,test_size])
    return train, dev, test
```
Does it make sense?

Comment (jcklie) [reply app_flair.py]: in sklearn you do not need to the labels. Your split is difficult to read, as you split from train twice, making it not a 80 10 10 split and checking the indices is quite hard.

Comment (bjtulinl) [reply app_flair.py]: Then maybe:
def dataset_split(list_sentences: list):
    l = len(list_sentences)
    dev_size: int = round(l / 10)
    test_size: int = round(l / 10)
    train, dev = random_split(list_sentences,[l-dev_size,dev_size])
    train, test = random_split(train, [train_l-test_size,test_size])
    return train, dev, test
Or do you think sklearn is a better choice?

Comment (bjtulinl) [reply app_flair.py]: Then maybe:
def dataset_split(list_sentences: list):
l = len(list_sentences)
dev_size: int = round(l / 10)
test_size: int = round(l / 10)
train, dev = random_split(list_sentences,[l-dev_size,dev_size])
train, test = random_split(train, [I-dev_size-test_size,test_size])
return train, dev, test
Or do you think sklearn is a better choice?

Comment (jcklie) [reply app_flair.py]: I just would use sklearn. Do you really need the test set in flair?

Comment (bjtulinl) [reply app_flair.py]: No. We can just set the corpus as:
    corpus = Corpus( train=train , dev=dev, test='')
"train" is used to train the model and "dev" is used to evaluate the score of the model. And the learning rate is only changed  when the f1 score of dev set decreases.

Comment (jcklie) [reply app_flair.py]: Then I would just use either torch or sklearn split once and ignore the test set. That should make it simpler.

**Comment by jcklie** [comment]
@bjtulinl It trains, but I am unable to get recommendations after the training. Does it work for you? I assume we need #20 .

**Comment by bjtulinl** [comment]
I will check it.

**Comment by jcklie** [comment]
I merge this and we will continue in #20 .
